### PR TITLE
WIP: Each tab has a stack navigator desktop

### PR DESCRIPTION
**Pull request by chrisnojima**
@keybase/react-hackers this applies these changes to desktop

### CODE REVIEW

**Comment by chrisnojima** [comment]
- [ ] check push

**Comment by buoyad** [comment]
Switching back to a tab always resets you to the root of the tab - intended?

**Comment by chrisnojima** [comment]
Nope will fix

**Comment by buoyad** [comment]
Navigating into a chat on another tab goes to the full inbox-and-conversation, I didn't expect to see the inbox

**Comment by chrisnojima** [comment]
That's on purpose

**Comment by buoyad** [comment]
Sure, just pointing out it was a weird experience for me

**Comment by chrisnojima** [comment]
fixed tab clear @buoyad

**Comment by chrisnojima** [comment]
- [ ] tab click to root

**Comment by buoyad** [comment]
The weird back behavior is inconsistent but here's one repro:

1. Fresh start app on people tab
2. Nav into a profile
3. Switch to teams tab
4. Hit back -> end up back on people tab

Sometimes you end up on the profile, sometimes on the root. If you skip step 2 step 4 doesn't nav anywhere.

File: shared/router-v2/header/index.desktop.js [hunk 1]
```diff
@@ -99,7 +99,7 @@ class Header extends React.PureComponent<Props, State> {
               type="iconfont-arrow-left"
               style={backArrowStyle}
               color={iconColor}
-              onClick={this.props.allowBack || this.props.loggedIn ? this.props.onPop : null}
```
Comment (buoyad) [inline shared/router-v2/header/index.desktop.js]: cc @cjb I removed this so the back button would be disabled at the root of the tab. Do we need this for something?

Comment (cjb) [reply shared/router-v2/header/index.desktop.js]: Nope!

**Pull request merged**